app/controllers/application_controller.rb

```diff
@@ -316,4 +320,12 @@ def mobile?
     client = DeviceDetector.new(request.user_agent)
     client.device_type != 'desktop'
   end
+
+  def x509_presented?
+    service_provider_request_from_session&.requested_attributes&.include?('x509_presented')
+  end
+
+  def force_piv_cac_setup?
+    Figaro.env.force_piv_cac_setup && aal3_policy.aal3_required? && x509_presented?
```

Comment:
Let's make this check for the feature flag more explicit -- as is, having an empty/nil value would still be truthu

```ruby
Figaro.env.force_piv_cac_setup == 'true'
```